cmd/libsnap-confine-private/infofile.c

```diff
@@ -76,6 +87,34 @@ int sc_infofile_get_key(FILE *stream, const char *key, char **value, sc_error **
         }
         /* Replace the trailing newline character with the NUL byte. */
         line_buf[nread - 1] = '\0';
+
+        /* Handle ini sections (if requested via non-null section name) */
+        if (line_buf[0] == '[') {
+            if (section == NULL) {
+                err = sc_error_init_simple("line %d contains unexpected section", lineno);
+                goto out;
+            }
+            section_matched = false;
+            char *start_section_name = line_buf + 1;
+            char *end_section_name = memchr(start_section_name, ']', nread - 2);
+            if (end_section_name == NULL) {
```

Comment:
Maybe we should expect section name to be kind of well formed after all, so the following check would be in place:
```
// expecting the closing ] to be present and be the last character in the line
if (end_section_name == NULL || end_section_name - start_section_name != nread - 3) {
    err = sc_error_init_simple("line %d is not a valid ini section", lineno);
    goto out;
}
```
This should catch inputs like `[section]]`, `[sec]tion`

Reply:
FWTW this helper is used for ini files generated by snapd, so we know they are valid, therefore I'd keep it simple for now. If we start to use it for arbitrary ini files, then these checks need more work, e.g.  the suggested change would fail on `[section]     \n`, we should check for `[]` etc.